### PR TITLE
Introduce get_headers and get_blocks methods

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

File: network-proto/node.proto [hunk 1]
```diff
@@ -27,6 +26,12 @@ message GetBlocksRequest {
     uint64 size = 3;
 }
 
+// Request to fetch the methods for blocks and headers.
+message DataRequest {
```
Comment (mzabaluev) [inline network-proto/node.proto]: I'd name this more descriptively. `BlockIds`?

File: network-proto/node.proto [hunk 2]
```diff
@@ -10,13 +10,12 @@ message TipRequest {}
...
-    // Block identifier of the chain tip.
-    bytes id = 1;
-    // Block date of the chain tip.
-    string blockdate = 2;
+    // Serialized content of the tip block header.
+    bytes blockheader = 1;
 }
 
 // Request message for methods GetBlocks and GetHeaders.
+// (reserved for the future work)
```
Comment (mzabaluev) [inline network-proto/node.proto]: This message is perhaps not needed and should be removed.
We can restore the fields from git history when about to introduce the new random access method `GetChain`.

File: network-proto/node.proto [hunk 3]
```diff
@@ -10,13 +10,12 @@ message TipRequest {}
 
 // Response message for method Tip.
 message TipResponse {
-    // Block identifier of the chain tip.
-    bytes id = 1;
-    // Block date of the chain tip.
-    string blockdate = 2;
+    // Serialized content of the tip block header.
+    bytes blockheader = 1;
```
Comment (mzabaluev) [inline network-proto/node.proto]: Use snake case, `block_header`?

File: network-core/src/server/block.rs [hunk 4]
```diff
@@ -64,6 +64,12 @@ pub trait BlockService {
     /// accepted by this node.
     fn tip(&mut self) -> Self::TipFuture;
 
+    /// Request to load list of blocks.
+    fn pull_blocks(&mut self, ids: &[Self::BlockId]) -> Self::PullBlocksFuture;
```
Comment (mzabaluev) [inline network-core/src/server/block.rs]: Please rename these to `get_blocks` and `get_headers` to be consistent with the protobuf method names.

File: network-proto/node.proto (hunk 3 above)
Comment (mzabaluev) [reply network-proto/node.proto]: `blockdate` was not snake case either, but it's almost like a single word.

**Comment by qnikst** [comment]
Please take another look

**Pull request merged**